File: web/tests/reference/prefix-sum/range-sum-query-immutable.js
class NumArray {
  constructor(nums) {
    this.prefix = new Array(nums.length + 1).fill(0);
    for (let i = 0; i < nums.length; i++) this.prefix[i + 1] = this.prefix[i] + nums[i];
  }

  sumRange(left, right) {
    return this.prefix[right + 1] - this.prefix[left];
  }
}

File: web/tests/reference/prefix-sum/range-sum-query-immutable.py
class NumArray:
    def __init__(self, nums):
        self.prefix = [0] * (len(nums) + 1)
        for i, x in enumerate(nums):
            self.prefix[i + 1] = self.prefix[i] + x

    def sumRange(self, left, right):
        return self.prefix[right + 1] - self.prefix[left]
